diff --git a/out/winPackager.js b/out/winPackager.js
--- a/out/winPackager.js
+++ b/out/winPackager.js
@@ -181,7 +181,12 @@ class WinPackager extends platformPackager_1.PlatformPackager {
         }
         const timer = (0, timer_1.time)("wine&sign");
         // rcedit crashed of executed using wine, resourcehacker works
-        if (process.platform === "win32" || process.platform === "darwin") {
+        if (process.platform === "win32") {
+            const vendor = await (0, windows_1.getRceditBundle)((_c = this.config.toolsets) === null || _c === void 0 ? void 0 : _c.winCodeSign);
+            const executable = arch === builder_util_1.Arch.ia32 ? vendor.x86 : vendor.x64;
+            await (0, builder_util_1.exec)(executable, args);
+        }
+        else if (process.platform === "darwin") {
             await (0, builder_util_1.executeAppBuilder)(["rcedit", "--args", JSON.stringify(args)], undefined /* child-process */, {}, 3 /* retry three times */);
         }
         else if (this.info.framework.name === "electron") {
diff --git a/templates/nsis/multiUser.nsh b/templates/nsis/multiUser.nsh
--- a/templates/nsis/multiUser.nsh
+++ b/templates/nsis/multiUser.nsh
@@ -28,14 +28,19 @@ Var installMode
       StrCpy $INSTDIR $perUserInstallationFolder
     ${else}
       StrCpy $0 "$LocalAppData\Programs"
-      System::Store S
-      # Win7 has a per-user programfiles known folder and this can be a non-default location
+      Push $1
+      Push $2
+      # UserProgramFiles can be a non-default location.
+      StrCpy $2 0
       System::Call 'SHELL32::SHGetKnownFolderPath(g "${FOLDERID_UserProgramFiles}", i ${KF_FLAG_CREATE}, p 0, *p .r2)i.r1'
       ${If} $1 == 0
-        System::Call '*$2(&w${NSIS_MAX_STRLEN} .s)'
-        StrCpy $0 $1
+        System::Call 'KERNEL32::lstrcpynW(w .r0, p r2, i ${NSIS_MAX_STRLEN})p'
+      ${endif}
+      # The API may allocate memory even on failure.
+      ${If} $2 != 0
         System::Call 'OLE32::CoTaskMemFree(p r2)'
       ${endif}
-      System::Store L
+      Pop $2
+      Pop $1
       StrCpy $INSTDIR "$0\${APP_FILENAME}"
     ${endif}
